xml/hardening.xml

```diff
@@ -3445,6 +3445,327 @@ Account Expires: Never</screen>
+        authentication process on local consoles to allow root only to login on
+        tty devices that are listed in the file <filename>/etc/securetty</filename>.
+      </para>
+     </step>
+     <step>
+      <para>
+       Remove all entries from <filename>/etc/securetty</filename> except one.
+       This limits the access to TTY devices for root.
+      </para>
+     <screen>#
+# This file contains the device names of tty lines (one per line,
+# without leading /dev/) on which root is allowed to login.
+#
+tty1</screen>
+     </step>
+     <step>
+      <para>
+       Check whether logins to other terminals will be rejected for
+       &rootuser;. A login on <literal>tty2</literal>, for example, should be
+       rejected right away without even the root password being queried.
+       Also make sure that you can still successfully login to
+       <literal>tty1</literal> and thus that &rootuser; is not locked out of
+       the system completely.
+      </para>
+     </step>
+    </procedure>
+
+    <important>
+     <para>
+      Avoid adding the <literal>pam_securetty</literal> PAM module to the
+      <filename>/etc/pam.d/common-auth</filename> file. This change would
+      break <command>su</command> and <command>sudo</command>, however,
+      because these tools would then also reject &rootuser; authentications.
+     </para>
+    </important>
+
+    <important>
+     <para>
+      These configuration changes will also cause root logins on serial
+      consoles like <filename>/dev/ttyS0</filename> to be denied. In case you
+      require such use cases you need to list the respective tty devices
+      additionally in the <filename>/etc/securetty</filename> file.
+     </para>
+    </important>
+   </sect2>
+
+   <sect2 xml:id="sec-sec-prot-retrict-root-graphical">
+    <title>Restricting Graphical Session Logins</title>
+
+    <para>
+     To improve security on your server, avoid to use graphical user
+     environments at all. Graphical programs are often
+     not designed to be run as &rootuser; and are more likely to
+     contain security issues than console programs. If you require a graphical
+     login, use a non-&rootuser; login. Configure your system to disallow
+     &rootuser; login on graphical session.
+    </para>
+
+    <para>
+     To deny the &rootuser; user login to graphical sessions you
+     can apply the same basic steps as outlined in
+     <xref linkend="sec-sec-prot-restrict-root-tty"/>.
+     You only need to also add the <literal>pam_securetty</literal> module to
+     the PAM stack file belonging to the display manager, for example
+     <filename>/etc/pam.d/gdm</filename> for GDM. The graphical session also
+     runs on a TTY device, <literal>tty7</literal>, by default. Therefore,
+     when you restrict &rootuser; logins to
+     <literal>tty1</literal>, then a &rootuser; login in the
+     graphical session will be denied.
```

Comment:
```suggestion
     To prevent &rootuser; from logging in to graphical sessions, you
     can apply the same basic steps as outlined in
     <xref linkend="sec-sec-prot-restrict-root-tty"/>.
     Just add the <literal>pam_securetty</literal> module to the PAM
     stack file belonging to the display manager&mdash;for example,
     <filename>/etc/pam.d/gdm</filename> for GDM. The graphical
     session also runs on a TTY device: by default, <literal>tty7</literal>.
     Therefore, if you restrict &rootuser; logins to <literal>tty1</literal>,
     then &rootuser; will be denied login in the graphical session.
```